plugins/woocommerce/src/Internal/Admin/Homescreen.php

```diff
@@ -49,7 +49,86 @@ public function __construct() {
+
+		if ( Features::is_enabled( 'shipping-smart-defaults' ) ) {
+			add_filter(
+				'woocommerce_admin_shared_settings',
+				array( $this, 'maybe_set_default_shipping_options_on_home' ),
+				9999
+			);
+		}
+	}
+
+	/**
+	 * Set free shipping in the same country as the store default
+	 * Flag rate in all other countries when any of the following conditions are ture
+	 *
+	 * - The store sells physical products, has JP and WCS installed and connected, and is located in the US.
+	 * - The store sells physical products, and is not located in US/Canada/Australia/UK (irrelevant if JP is installed or not).
+	 * - The store sells physical products and is located in US, but JP and WCS are not installed.
+	 *
+	 * @param array $settings shared admin settings.
+	 * @return array
+	 */
+	public function maybe_set_default_shipping_options_on_home( $settings ) {
+		$current_screen = get_current_screen();
+
+		// Abort if it's not the homescreen.
+		if ( ! isset( $current_screen->id ) || 'woocommerce_page_wc-admin' !== $current_screen->id ) {
+			return $settings;
+		}
+
+		// Abort if we already created the shipping options.
+		$already_created = get_option( 'woocommerce_admin_created_default_shipping_zones' );
+		if ( 'yes' === $already_created ) {
+			return $settings;
+		}
+
+		$user_skipped_obw = $settings['onboarding']['profile']['skipped'] ?? false;
+		$store_address = $settings['preloadSettings']['general']['store_address'] ?? '';
+		$product_types = $settings['onboarding']['profile']['product_types'] ?? array();
+
+		// If user skipped the obw or has not completed the store_details
+		// then we assume the user is going to sell physical products.
+		if ( $user_skipped_obw || '' === $store_address ) {
+			$product_types[] = 'physical';
+		}
+
+		if ( false === in_array( 'physical', $product_types, true ) ) {
+			return $settings;
+		}
+
+		$country_code = wc_format_country_state_string( $settings['preloadSettings']['general']['woocommerce_default_country'] )['country'];
+		$country_name = WC()->countries->get_countries()[ $country_code ] ?? null;
+
+		// we also need to make sure woocommerce_store_address is not empty
+		// to make sure store country is set by an actual user
+		// since woocommerce_store_address is set to US:CA by default.
+		if ( '' === $country_code || null === $country_name || '' === $store_address ) {
+			return $settings;
+		}
+
+		$is_jetpack_installed = in_array( 'jetpack', $settings['plugins']['installedPlugins'] ?? array(), true );
+		$is_jetpack_connected = $settings['dataEndpoints']['jetpackStatus']['isUserConnected'] ?? false;
+		$is_wcs_installed     = in_array( 'woocommerce-services', $settings['plugins']['installedPlugins'] ?? array(), true );
+
+		if (
+			( 'US' === $country_code && $is_jetpack_installed && $is_wcs_installed && $is_jetpack_connected )
+			||
+			( ! in_array( $country_code, array( 'US', 'CA', 'AU', 'UK' ), true ) )
```

Comment:
I found that we should use `GB` for `United Kingdom`. I'll also need to fix it for my other merged PRs. 😅 

https://github.com/woocommerce/woocommerce/blob/14066bdeae9abdec059b2a647489a07d7e7b3d9f/plugins/woocommerce/i18n/countries.php#L249

Reply:
Thank you! Updated in https://github.com/woocommerce/woocommerce/pull/33547/commits/67c8a937993f8c79b5c3df0eb32231050ed52260